m2cgen/interpreters/interpreter.py

```diff
@@ -24,7 +24,15 @@ def interpret(self, expr):
     def _pre_interpret_hook(self, expr, **kwargs):
         return None, kwargs
 
-    def _do_interpret(self, expr, to_reuse=None, left_precedence=None, right_precedence=None, **kwargs):
+    def _do_interpret(
+        self,
+        expr,
+        to_reuse=None,
+        left_precedence=None,
+        right_precedence=None,
+        right_is_associative=None,
```

Comment:
I'm not sure about the naming here. Whether it's `right` or `left` doesn't really impact this particular parameter (unlike in case of `precedence`). I'd instead call it `parent_is_associative` or something.

Additionally, now I'm not quite sure about `*_precedence` names either. Maybe it'd be less confusing if we have something like this:
1. `parent_precedence` - pretty self-explanatory.
2. `is_right_from_parent` / `is_left_from_parent` - a boolean flag to indicate whether the current is the in the right  or left position.
3. `parent_is_associative` / `is_parent_associative` - the new parameter. 

What do you think?

Reply:
I like your proposal for `is_parent_associative` name.

But for `#1` and `#2` items I can't get the idea, unfortunately. Could you please provide a small code snippet with an example for these two params will be used in the wild?

Reply:
Sure. Something like this:
```
def _wrap_infix_expr(...):
    if parent_precedence:
        if is_left_from_parent:
            return parent_precedence > expr.precedence
        return parent_precedence > expr.precedence or (parent_precedence == expr.precedence and is_parent_associative)
```